### DISCUSSION TITLE
lwIP_enc28j60

### DISCUSSION THREAD

**Discussion by francismariano** [Networking-for-Arduino/EthernetENC#28] [General]
Hello,

In the readme I see the following statement: 
"For esp8266 Arduino there is a much better option for enc28j60 than using this library. Use the lwIP_enc28j60 library bundled with esp8266 Arduino boards support package 3."

Can you please give me more details about that? Do you know some guide explaining how to use lwIP_enc28j60 for ESP8266?

Thank you in advanced.

**Comment by JAndrassy** [comment]
see the example for the lwIP_w5500 library

**Reply by rescalda** [reply]
**francismariano**: did you find the above example? I cannot find any!
**jandrassy**: Could you be so kind to indicate where the _example for the lwIP_w5500 library is? In the folder in ESP8266 3.0.2 board library there are none!
Thanks

**Reply by JAndrassy** [reply]
in Arduino IDE Examples menu

**Reply by rescalda** [reply]
**jandrassy:** In the "Arduino IDE Examples menu", under ESP8266 folder, I can find the subfolder _Ethernet_ that links to many examples using **#include <Ethernet.h>**. Should I replace _Ehternet_ with _ENC28J60lwIP_ I get compile errors like _error: 'EthernetClient' does not name a type; did you mean 'EthernetClass'?_.
Under _Examples for Custom Libraries_, I don't find anything related to ENC28J60lwIP. The only one folder with lwIP in the name il lwIp_PPP but, of course, it has nothing to do with ethernet.

**Reply by JAndrassy** [reply]
sorry I see there is some problem in IDE and the example doesn't show there.
https://github.com/esp8266/Arduino/blob/master/libraries/lwIP_w5500/examples/TCPClient/TCPClient.ino

**Reply by rescalda** [reply]
Ah! **This works fine!** Many, many, thanks!

**Reply by ghost** [reply]
Hi,

Sorry for updating an old post but i've an issue with the same hardware :
ESP8266 (WEMOS LOLIN D1 Mini) and ENC28J60

if i use the EthernetENC library the DHCP response never come.

After that i saw the important note on the main page of this librairie, but i don't understand what done.
fem moment later i saw this post but the exemple does not exist anymore.

If i use this exemple :
https://github.com/esp8266/Arduino/blob/master/libraries/lwIP_enc28j60/src/utility/enc28j60.cpp

by using 
    ArduinoENC28J60lwIP Ethernet(2);
DHCP does answer also

Can you help me please ?

**Reply by JAndrassy** [reply]
https://github.com/esp8266/Arduino/blob/master/libraries/lwIP_Ethernet/examples/EthClient/EthClient.ino
